gpustack/worker/backend_dependency_manager.py

```diff
@@ -0,0 +1,126 @@
+import logging
+from typing import Dict, List
+from dataclasses import dataclass
+
+from gpustack.schemas.models import BackendEnum
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class BackendDependencySpec:
+    """
+    Represents a backend dependency specification.
+
+    Attributes:
+        backend: The backend name (e.g., 'vox-box', 'vllm')
+        dependencies: List of dependency specifications (e.g., ['transformers==4.51.3', 'torch>=2.0.0'])
+    """
+
+    backend: str
+    dependencies: List[str]
+
+    def to_pip_args(self) -> str:
+        """
+        Convert dependencies to pip arguments format.
+
+        Returns:
+            String in format "--pip-args='dep1 dep2 dep3'"
+        """
+        if not self.dependencies:
+            return ""
+
+        deps_str = " ".join(self.dependencies)
+        return f"--pip-args='{deps_str}'"
+
+
+class BackendDependencyManager:
+    """
+    Manages backend dependencies for different inference backends.
+
+    Examples:
+        - model_env: {"GPUSTACK_BACKEND_DEPS"="transformers==4.53.3,torch>=2.0.0"}
+    """
+
+    def __init__(self, backend: str, version: str, model_env: Dict[str, str] = None):
+        self.backend = backend
+        self.version = version
+        self._custom_specs: BackendDependencySpec = None
+
+        # Initialize default dependencies for each backend
+        self.default_dependencies_specs: Dict[str, Dict[str, List[str]]] = {
+            BackendEnum.VOX_BOX: {
+                "v0.0.20": ["transformers==4.51.3"],
+                "v0.0.19": ["transformers==4.51.3"],
+                "v0.0.18": ["transformers==4.51.3"],
+            },
+            BackendEnum.VLLM: {
+                "v0.10.0": ["transformers==4.53.3"],
+                "v0.9.2": ["transformers==4.53.3"],
+                "v0.9.1": ["transformers==4.53.3"],
+            },
```

Comment:
Can we use https://packaging.pypa.io/en/stable/specifiers.html to define something like "<=v0.10.0": ["transformers==4.53.3"]